bokehjs/src/lib/models/annotations/html/title.ts

```diff
@@ -0,0 +1,153 @@
+import * as p from "core/properties"
+
+export class TitleView extends TextAnnotationView {
+  override model: Title
+  override visuals: Title.Visuals
+  override layout: Layoutable
+  override panel: Panel
+
+  protected _get_location(): [number, number] {
+    const hmargin = this.model.offset
+    const vmargin = this.model.standoff/2
+
+    let sx: number, sy: number
+    const {bbox} = this.layout
+    switch (this.panel.side) {
+      case "above":
+      case "below": {
+        switch (this.model.vertical_align) {
+          case "top":    sy = bbox.top     + vmargin; break
+          case "middle": sy = bbox.vcenter;           break
+          case "bottom": sy = bbox.bottom  - vmargin; break
+        }
+
+        switch (this.model.align) {
+          case "left":   sx = bbox.left    + hmargin; break
+          case "center": sx = bbox.hcenter;           break
+          case "right":  sx = bbox.right   - hmargin; break
+        }
+        break
+      }
+      case "left": {
+        switch (this.model.vertical_align) {
+          case "top":    sx = bbox.left    + vmargin; break
+          case "middle": sx = bbox.hcenter;           break
+          case "bottom": sx = bbox.right   - vmargin; break
+        }
+
+        switch (this.model.align) {
+          case "left":   sy = bbox.bottom  - hmargin; break
+          case "center": sy = bbox.vcenter;           break
+          case "right":  sy = bbox.top     + hmargin; break
+        }
+        break
+      }
+      case "right": {
+        switch (this.model.vertical_align) {
+          case "top":    sx = bbox.right   - vmargin; break
+          case "middle": sx = bbox.hcenter;           break
+          case "bottom": sx = bbox.left    + vmargin; break
+        }
+
+        switch (this.model.align) {
+          case "left":   sy = bbox.top     + hmargin; break
+          case "center": sy = bbox.vcenter;           break
+          case "right":  sy = bbox.bottom  - hmargin; break
+        }
+        break
+      }
+    }
+
+    return [sx, sy]
+  }
+
+  protected _render(): void {
+    const {text} = this.model
+    if (text == null || text.length == 0)
+      return
+
+    this.model.text_baseline = this.model.vertical_align
+    this.model.text_align = this.model.align
+
+    const [sx, sy] = this._get_location()
+    const angle = this.panel.get_label_angle_heuristic("parallel")
+
+    this._paint(this.layer.ctx, text, sx, sy, angle)
+  }
+
+  // XXX: this needs to use CSS computed styles
+  protected override _get_size(): Size {
+    const {text} = this.model
+    const graphics = new TextBox({text})
+    graphics.visuals = this.visuals.text.values()
+    const {width, height} = graphics.size()
+    // XXX: The magic 2px is for backwards compatibility. This will be removed at
+    // some point, but currently there is no point breaking half of visual tests.
+    return {width, height: height == 0 ? 0 : 2 + height + this.model.standoff}
```

Comment:
could we not just add 2 to the default `standoff` value, to remove the magic number here w/o breaking tests?

Reply:
This particular logic is completely wrong for an HTML annotation. However, the same code is in canvas `Title`, and it will be removed soon enough. Though I need to review this logic from grounds up, because the way this annotation works doesn't add up.